comit/src/float_math.rs

```diff
@@ -0,0 +1,223 @@
+use bitcoin::hashes::core::cmp::Ordering;
+use num::{BigUint, Zero};
+use std::str::FromStr;
+
+/// Truncate the float's mantissa to length `precision`.
+pub fn truncate(float: f64, precision: u16) -> f64 {
+    let s = float.to_string();
+    truncate_str(&s, precision)
+}
+
+/// Truncate the float's mantissa to length `precision`.
+pub fn truncate_str(float: &str, precision: u16) -> f64 {
+    let mut string = float.to_string();
```

Comment:
Because this accepts a string, there is no guarantee the string actually represents a number.
We should check that and also return a result.
Also, we should proptest this function with any arbitrary string value and ensure it does not panic/overflows.

Reply:
Oh, you mean we should actually write the API properly instead of me just hacking ups some shit and trying to merge it into the comit lib - if this is what you mean you are totally correct. Sloppy work here from me. I think we should move comit to a separate crate and enforce much stricter code practices on it.